pkg/webhooks/podv1mutator/sidecar.go

```diff
@@ -125,9 +138,69 @@ func (esc *envoySidecarConfig) PopulateFromAnnotations(annotations map[string]st
 	esc.clientCertSecret = getStringParam(paramClientCertificate, annotations)
 	esc.extraArgs = getStringParam(paramEnvoyExtraArgs, annotations)
 
+	resources, err := getContainerResourceRequirements(annotations)
+	if err != nil {
+		return err
+	}
+	esc.resources = resources
+
 	return nil
 }
 
+func lookupMarin3rAnnotation(key string, annotations map[string]string) (string, bool) {
+	value, ok := annotations[fmt.Sprintf("%s/%s", marin3rAnnotationsDomain, key)]
+	return value, ok
+}
```

Comment:
Would be nice to use this small helper for all the code in this file WDYT?

Reply:
I agree.
I refactored the sidecar.go code to use it and I added the changes as a new commit